### PR TITLE
Make this rails app React ready

### PR DESCRIPTION
**Pull request by bunnymatic**
* Add webpacker gem and run webpacker:install
* Add jest & enzyme for javascript testing
* Add dummy react component (ClickMe) with tests
* Put the dummy component on the root page to prove that it works
* Add component registry and `react_component` application helper (like what `react-rails` gives you)

### CODE REVIEW

**Comment by bunnymatic** [comment]
I expect some review work and discussion here.  I've cribbed this work from a recent project on which I started with the default raygun template and put in the basics for React/ES6 work.  If we wanted to push this further, I might also think about ripping jQuery out to encourage the use of only one Javascript framework/toolset.

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -36,4 +36,20 @@ def alert_class(alert_type)
     }.with_indifferent_access.fetch(alert_type, alert_type.to_s)
     "alert-#{alert_type}"
   end
+  def react_component(component_name, props={}, html_options = {}, render_order: 0)
```
Comment (eskfung) [inline app/helpers/application_helper.rb]: I found myself gravitating towards https://github.com/renchap/webpacker-react to have this view helper.

Comment (bunnymatic) [inline app/helpers/application_helper.rb]: That totally works.  I recall some helper like this for angular for which someone had written a Gem but then changes to Sprockets messed it up.  it seemed about as simple to maintain your own helper method as to figure out that the Gem is not maintained or needs an update.

**Comment by bentona** [comment]
@bunnymatic @christiannelson Starting a new rails project and would love to use this since I'm not too familiar with how react & rails work together - anything I can do to help?

**Comment by bunnymatic** [comment]
clearly it's been forever since i started this little branch.  

At this point, you should be able to use (or it's already there) the `webpacker` gem which at it's current version is miles from where this branch is.

If you want to chat about it we can.  Though you might also check in with @mattbrictson and see where SprayGun is.  I think he had a short instruction list to "run spray gun and hook it to a rails app".  and that might take the place of this PR/branch.

**Comment by mattbrictson** [comment]
@bentona check out this doc: [Using spraygun-react with a Rails backend](https://github.com/carbonfive/spraygun-react/blob/master/docs/how-to-use-with-rails-backend.md)

It is not completely finished though: in particular it is pretty tricky to get capybara feature specs to work. I believe @marclove came up with a solution for this on his current project.

**Comment by bentona** [comment]
@mattbrictson @bunnymatic this is great, I didn't even know spraygun was a thing! Thank you.

**Comment by bunnymatic** [comment]
stale!

**Pull request closed without merging**